composer/callbacks/speed_monitor.py

```diff
@@ -124,3 +179,15 @@ def batch_end(self, state: State, logger: Logger):
     def eval_end(self, state: State, logger: Logger):
         del logger  # unused
         self.total_eval_wct += state.eval_timestamp.total_wct.total_seconds()
+        assert state.dataloader_label is not None, 'evaluator label must not be None'
+        if state.dataloader_label not in self.eval_wct_per_label:
+            self.eval_wct_per_label[state.dataloader_label] = []
+        self.eval_wct_per_label[state.dataloader_label].append(state.eval_timestamp.total_wct.total_seconds())
+        max_dur = self.get_elapsed_duration(state)
+        if max_dur is None:
+            warnings.warn(
+                'Attempting to estimate remaining time but `max_duration` is not set. Skipping adjustment for evaluation time.'
+            )
+        else:
+            self.eval_rate_per_label[state.dataloader_label] = max_dur / len(
```

Comment:
Wait I'm a bit confused as to how this is computed. So it's elapsed_duration, which is of units batch? and it's divided by `len(self.eval_wct_per_label[state.dataloader_label])`, which is the number of batches that had an eval? Any chance we could add another comment here or give `len(self.eval_wct_per_label[state.dataloader_label])` a descriptive variable name?

Reply:
So this is basically saying elapsed duration divided by number of calls. The goal is to compute frequency of eval calls in duration units.

Reply:
ok yeah this makes sense now that I know elapsed_duration is a fraction